### PR TITLE
Update Zerl version and client build system

### PR DESCRIPTION
**Pull request by z-silver**
This should make things Just Work<sup>TM</sup> for the most part. It also eliminates some kludges and generally streamlines things.

How to test: run `just client` inside the flake. Maybe run the server as well and connect to it if you're feeling particularly paranoid.

### CODE REVIEW

**Review by mtrsk — APPROVED** [review]

File: client/build.zig [hunk 1]
```diff
@@ -3,7 +3,7 @@ const std = @import("std");
 // Although this function looks imperative, note that its job is to
 // declaratively construct a build graph that will be executed by an external
 // runner.
-pub fn build(b: *std.Build) void {
+pub fn build(b: *std.Build) !void {
```
Comment (EduardoLR10) [inline client/build.zig:6]: I'm not a huge fan of things named `a`, `b`, `c`, etc. Is this a Zig convention that I'm not aware or we can do better?

**Review by EduardoLR10 — CHANGES_REQUESTED** [review]
I would like to have a list of things in the PR description that I should test locally to verify all the changes. Is this `just client` enough within Nix or something else is needed?

File: client/build.zig (hunk 1 above)
Comment (z-silver) [reply client/build.zig:6]: It is conventional to name the build object `b` in your build functions, yes. It's even like that in the default `build.zig` you get when you `zig init` a new project.

**Comment by z-silver** [comment]
> I would like to have a list of things in the PR description that I should test locally to verify all the changes. Is `just client` enough within Nix or is something else needed?

That should be enough, but you can also run the backend and try connecting to it if you're feeling paranoid. (I updated the PR description as well.)

**Review by EduardoLR10 — APPROVED** [review]

**Pull request merged**